src/hooks/transactions/useSignMultipleTransactions.tsx

```diff
@@ -109,9 +113,15 @@ export function useSignMultipleTransactions({
     const { tokenId } = transactionTokenInfo;
     const receiver = transaction.getReceiver().toString();
     const sender = transaction.getSender().toString();
-    const isLoggedInWithDifferentAccount = sender && address !== sender;
+    const senderAccount = await getAccount(sender);
+    const isLoggedInWithDifferentAccount = checkIsInvalidSender(
+      senderAccount,
+      address
+    );
 
-    // Prevent signing transactions with different account
+    // Don't allow signing if the sender from the transaction
```

Comment:
Don't allow signing if the logged in account's address is neither the sender or the sender account's active guardian